sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs

```diff
@@ -353,49 +355,49 @@ public void TearDownRecordedTestClass()
             if (Mode == RecordedTestMode.Record)
             {
                 var testClassDirectory = new DirectoryInfo(GetSessionFileDirectory());
-                if (!testClassDirectory.Exists)
+                if (testClassDirectory.Exists)
                 {
-                    return;
-                }
-
-                var knownMethods = new HashSet<string>();
+                    var knownMethods = new HashSet<string>();
 
-                // Management tests record in ctor
-                knownMethods.Add(GetType().Name);
+                    // Management tests record in ctor
+                    knownMethods.Add(GetType().Name);
 
-                // Collect all method names
-                foreach (var method in GetType()
-                             .GetMethods(BindingFlags.Public | BindingFlags.Static | BindingFlags.Instance))
-                {
-                    // TestCase attribute allows specifying a test name
-                    foreach (var attribute in method.GetCustomAttributes(true))
+                    // Collect all method names
+                    foreach (var method in GetType()
+                                 .GetMethods(BindingFlags.Public | BindingFlags.Static | BindingFlags.Instance))
                     {
-                        if (attribute is ITestData { TestName: { } name })
+                        // TestCase attribute allows specifying a test name
+                        foreach (var attribute in method.GetCustomAttributes(true))
                         {
-                            knownMethods.Add(name);
+                            if (attribute is ITestData { TestName: { } name })
+                            {
+                                knownMethods.Add(name);
+                            }
                         }
-                    }
-
-                    knownMethods.Add(method.Name);
-                }
 
-                foreach (var fileInfo in testClassDirectory.EnumerateFiles())
-                {
-                    bool used = knownMethods.Any(knownMethod => fileInfo.Name.StartsWith(knownMethod, StringComparison.CurrentCulture));
+                        knownMethods.Add(method.Name);
+                    }
 
-                    if (!used)
+                    foreach (var fileInfo in testClassDirectory.EnumerateFiles())
                     {
-                        try
-                        {
-                            fileInfo.Delete();
-                        }
-                        catch
+                        bool used = knownMethods.Any(knownMethod => fileInfo.Name.StartsWith(knownMethod, StringComparison.CurrentCulture));
+
+                        if (!used)
                         {
-                            // Ignore
+                            try
+                            {
+                                fileInfo.Delete();
```

Comment:
rather than having the TestFramework delete these, would it be better to log a warning if we detect that the assets are migrated to the proxy test recording repo and tell them it is safe to delete them manually?

Reply:
This is existing code for deleting unused recordings. I think it was just refactored to remove one level of nesting, so the diff is misleading.

Reply:
> This is existing code for deleting unused recordings. I think it was just refactored to remove one level of nesting, so the diff is misleading.

That's correct. Not part of this PR.